ibm/service/cos/resource_ibm_cos_bucket_objectlock_configuration.go

```diff
@@ -0,0 +1,348 @@
+package cos
+
+import (
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/IBM-Cloud/terraform-provider-ibm/ibm/conns"
+	"github.com/IBM-Cloud/terraform-provider-ibm/ibm/flex"
+	"github.com/IBM-Cloud/terraform-provider-ibm/ibm/validate"
+	"github.com/IBM/ibm-cos-sdk-go/aws"
+	"github.com/IBM/ibm-cos-sdk-go/service/s3"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func ResourceIBMCOSBucketObjectlock() *schema.Resource {
+	return &schema.Resource{
+		Create:   resourceIBMCOSBucketObjectlockCreate,
+		Read:     resourceIBMCOSBucketObjectlockRead,
+		Update:   resourceIBMCOSBucketObjectlockUpdate,
+		Delete:   resourceIBMCOSBucketObjectlockDelete,
+		Importer: &schema.ResourceImporter{},
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(60 * time.Minute),
+			Update: schema.DefaultTimeout(20 * time.Minute),
+			Delete: schema.DefaultTimeout(10 * time.Minute),
+		},
+		Schema: map[string]*schema.Schema{
+			"bucket_crn": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "COS bucket CRN",
+			},
+			"bucket_location": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "COS bucket location",
+			},
+			"endpoint_type": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ValidateFunc: validate.ValidateAllowedStringValues([]string{"public", "private", "direct"}),
+				Description:  "COS endpoint type: public, private, direct",
+				Default:      "public",
+			},
+			"object_lock_configuration": {
+				Type:        schema.TypeList,
+				Required:    true,
+				MaxItems:    1,
+				Description: "Bucket level object lock settings includes Days, Years, Mode.",
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"objectlockenabled": {
+							Type:        schema.TypeString,
```

Comment:
can we reame variable with "_"
objectlockenabled to object_locke_nabled

Reply:
same for down all arguments names which have multiple words needs to be seperated by "_"
defaultretention -> default_retention

Reply:
addressed the comment